tensorrt_llm/_torch/disaggregation/native/messenger.py

```
@@ -118,10 +121,20 @@ def start(self) -> None:
     def send(self, messages: list[bytes], recipient: Optional[bytes] = None) -> None:
-        if recipient:
-            self._socket.send_multipart([recipient] + messages)
-        else:
-            self._socket.send_multipart(messages)
+        """Send one multipart message. Safe to call from any thread.
+
+        pyzmq sockets are not thread-safe: concurrent send_multipart() interleaves
+        frames, so peers decode misaligned payloads. Owning the serialization here
+        means no caller can get it wrong, from any thread.
+        """
+        frames = [recipient] + messages if recipient else messages
```

Ruff (0.16.2)

[warning] Consider `[recipient, *messages]` instead of concatenation

Replace with `[recipient, *messages]`

(RUF005)